lib/src/gradle_parser.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:petitparser/petitparser.dart';
+
+/// Grammar for Gradle
+class GradleGrammerDefinition extends GrammarDefinition {
+  @override
+  Parser start() => ref0(value).end();
+
+  /// A token, smallest unit to parse
+  Parser token(Object source, [String name]) {
+    if (source is String) {
+      return source.toParser(message: 'Expected ${name ?? source}').trim();
+    } else if (source is Parser) {
+      ArgumentError.checkNotNull(name, 'name');
+      return source.flatten('Expected $name').trim();
+    } else {
+      throw ArgumentError('Unknown token type: $source.');
+    }
+  }
+
+  /// Prodctions to parse a block of elements
+  Parser block() => ref1(token, '{') & ref0(elements) & ref1(token, '}');
+
+  /// A list of key-value pairs seperated by a new line
+  Parser elements() =>
+      ref0(pair).separatedBy(ref1(token, '\n'), includeSeparators: false);
+
+  /// Proudctions to parse each block
+  Parser members() =>
+      ref0(pair).separatedBy(ref1(token, '\n'), includeSeparators: false);
+
+  /// a key-value pair seperated by a whitespace
+  Parser pair() => ref0(stringToken) & ref1(token, ' ') & ref0(value);
+
+  /// Primitive value tokens
+  Parser trueToken() => ref1(token, 'true');
+  Parser falseToken() => ref1(token, 'false');
+  Parser stringToken() => ref2(token, ref0(stringPrimitive), 'string');
+  Parser numberToken() => ref2(token, ref0(numberPrimitive), 'number');
+
+  /// Possible values for parsing
+  Parser value() => [
+        ref0(members),
+        ref0(block),
+        ref0(stringToken),
+        ref0(numberToken),
+        ref0(trueToken),
+        ref0(falseToken),
+      ].toChoiceParser(failureJoiner: selectFarthestJoined);
+
+  Parser characterPrimitive() => pattern('^"\\');
+  Parser stringPrimitive() =>
+      char('"') & ref0(characterPrimitive).star() & char('"');
+  Parser numberPrimitive() => (digit().plus());
+}
+
+/// Parser for Gradle
+class GradleParserDefinition extends GradleGrammerDefinition {
```

Comment:
It might be overkill to not just add all the production actions in `GradleGrammerDefinition`...

Actually, it might even be simpler to just do simple grammar without a custom subclass of `GrammarDefinition`.. On the other hand I can see it as an elegant way to encapsulate things.. So long as the monstrosity is private -- as a type `GrammarDefinition` has a lot of members that have no business being publicly re-exported from a library -- IMO, it only make sense as a private type.

https://pub.dev/packages/petitparser#writing-a-simple-grammar